app/controllers/profiles_controller.rb

```diff
@@ -0,0 +1,61 @@
+class ProfilesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :complete_profile, only: [:show, :edit]
+  before_action :assign_profile, only: [:show, :edit, :update]
+
+  def show
+  end
+
+  def new
+    redirect_to(profile_path) && return if profile.present?
+    @profile = Profile.new
+  end
+
+  def create
+    @profile = Profile.new(profile_params)
+    @profile.user = current_user
+    if @profile.save
+      redirect_to(profile_path, notice: 'Thanks! Your profile has successfully been saved')
+    else
+      flash.alert = 'Please fill in the required fields'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.